components/molecules/ContributorProfileHeader/contributor-profile-header.tsx

```diff
@@ -327,4 +329,79 @@ const ContributorProfileHeader = ({
   );
 };
 
+// Making this dropdown seperate to optimize for performance and not fetch certain data until the dropdown is rendered
+const AddToListDropdown = ({ username }: { username: string }) => {
+  const [selectListOpen, setSelectListOpen] = useState(false);
+  const [selectedList, setSelectedList] = useState<OptionKeys[]>([]);
+  const { data } = useFetchAllLists();
+  const { data: contributor } = useFetchUser(username ?? "");
+  const { toast } = useToast();
+
+  const listOptions = data ? data.map((list) => ({ label: list.name, value: list.id })) : [];
+
+  const handleSelectList = (value: OptionKeys) => {
+    const isOptionSelected = selectedList.some((s) => s.value === value.value);
+    if (isOptionSelected) {
+      setSelectedList((prev) => prev.filter((s) => s.value !== value.value));
+    } else {
+      setSelectedList((prev) => [...prev, value]);
+    }
+  };
+
+  const handleAddToList = async () => {
+    if (selectedList.length > 0 && contributor) {
```

Comment:
We could also fetch the contributor ID from GitHub if they are not in our data at all. This is good for now